### PR TITLE
Add ignore_index to dice, focal and tversky loss

### LINKED ISSUE #839: "ignore_index" argment for focal loss and dice loss
When I try to train for segmentation on VOC 2012 with Dice loss, I always run into cuda illegal error and I finally found that this may be caused by out-of-range indexes (255 out of 20 classes). Since pytorch CrossEntropyLoss has an argment "ignore_index" to deal with this issue, I wonder if there is a proper way to deal with uncertain indexes?

### PR DESCRIPTION
**Pull request by falckt**
#### Changes
* adds `ignore_index` argument to dice and tversky losses. 
* a second commit also adds code to the focal loss, but without tests. I couldn't easily extend an existing test case to properly test the new parameter. IMHO, I would drop this second commit for now.

fixes #839

#### Type of change
- [x] 🧪 Tests Cases
- [x] 🔬 New feature (non-breaking change which adds functionality)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by shijianjian — APPROVED** [review]
I think it is an important feature but missed everywhere in many loss functions. Is it possible to apply this on other losses?

**Comment by falckt** [comment]
I initially tried adding it to all the semantic segmentation losses, but the logic needed for the lovasz losses would need to be different. Unfortunately I don't have the time to dig into them and see how to implement it properly.

I did add basic test coverage for the focal losses though.

File: kornia/losses/focal.py [hunk 1]
```diff
@@ -20,6 +20,7 @@ def focal_loss(
     gamma: float = 2.0,
     reduction: str = "none",
     weight: Optional[Tensor] = None,
+    ignore_index: int = -100,
```
Comment (edgarriba) [inline kornia/losses/focal.py:23]: can this be set as None too ?

File: kornia/losses/tversky.py [hunk 2]
```diff
@@ -66,9 +72,23 @@ def tversky_loss(
     # compute softmax over the classes axis
     pred_soft: torch.Tensor = F.softmax(pred, dim=1)
 
+    # check if all pixels are valid
+    valid_idx = target != ignore_index
```
Comment (edgarriba) [inline kornia/losses/tversky.py:76]: maybe this can be wrapped into a reusable function across all the losses

**Review by edgarriba — APPROVED** [review]
LGTM. Minor comment that can be improved in a further PR

File: kornia/losses/focal.py (hunk 1 above)
Comment (falckt) [reply kornia/losses/focal.py]: I initially followed value and typing from pytorch, e.g.
https://pytorch.org/docs/stable/generated/torch.nn.functional.cross_entropy.html

considering that the kornia losses are pure python, I added a short circuit for None that saves some computation. I left the default value the same as in pytorch though. It is not a breaking change as -100 would crash if fed into one_hot. As such no existing code should have it as a valid label.

File: kornia/losses/tversky.py (hunk 2 above)
Comment (falckt) [reply kornia/losses/tversky.py]: I pulled it out into a helper function, not sure where to place it though. Any suggestions appreciated.

Comment (edgarriba) [reply kornia/losses/tversky.py]: This sounds good

File: kornia/losses/_utils.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from typing import Optional
```
Comment (edgarriba) [inline kornia/losses/_utils.py:1]:
```suggestion
from __future__ import annotations
from typing import Optional
```
This will fix the tests

**Review by edgarriba — APPROVED** [review]
